\subsection{Detailed version of the additive benchmark}

\begin{table}[H]
    \centering
    \includegraphics[width=\textwidth]{./supplementaries/scprint_2_media/image4.png}
    \caption[Detailed version of the additive benchmark]{, listing every value.}
    \label{table-s1-detailed-version-of-the-additive-benchmark}
\end{table}

\subsection{Detailed scIB biological conservation scores on the xenium dataset}

\begin{table}[H]
    \centering
    \includegraphics[width=\textwidth]{./supplementaries/scprint_2_media/image30.png}
    \caption[Detailed scIB biological conservation scores on the xenium dataset.]{scPRINT vs PCA on expression. ScPRINT performs better, likely by denoising the expression.}
    \label{table-s3-detailed-scib-biological-conservation-scores-on-the-xenium-dataset}
\end{table}

\subsection{Detailed scIB scores on the unseen species integration task}

\begin{table}[H]
    \centering
    \includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image20.png}
    \caption[Detailed scIB scores on the unseen species integration task.]{Full results comparing no integration, random embeddings sampled from the multivariate Gaussian, and different versions of scPRINT zero-shot or fine-tuned, using the merged embeddings or the cell type ones.}
    \label{table-s4-detailed-scib-scores-on-the-unseen-species-integration-task}
\end{table}

\section{Supplementary figures for scPRINT-2}\label{supplementary-figures-2}

\subsection{Illustration of the full scPRINT-2's architecture, input, and output}

\begin{figure}[H]
    \centering
    \includegraphics[width=5.92104in,height=6.85698in]{./supplementaries/scprint_2_media/image24.png}
    \caption[Illustration of the full scPRINT-2's architecture, input, and output.]{In-depth view with all its main different components and the data flow.}
    \label{fig-s1-illustration-of-the-full-scprint-2s-architecture-input-and-output}
\end{figure}

\subsection{Barplot of the F1-macro scores on the label-projection task of the Open Problem benchmark}

\begin{figure}[H]
    \centering
    \includegraphics[width=6.22396in,height=4.54658in]{./supplementaries/scprint_2_media/image17.png}
    \caption[Barplot of the F1-macro scores on the label-projection task of the Open Problem benchmark.]{Comparison of scPRINT-1 and scPRINT-2, zero-shot and finetuned, with all other tested methods.}
    \label{fig-s2-barplot-of-the-f1-macro-scores-on-the-label-projection-task-of-the-open-problem-benchmark}
\end{figure}

\subsection{Heatmap of ethnicity prediction relationship across samples}

\begin{figure}[H]
    \centering
    \includegraphics[width=6.39163in,height=6.39163in]{./supplementaries/scprint_2_media/image1.png}
    \caption[Heatmap of ethnicity prediction relationship across samples.]{Generated using labels predicted as top-1 (x-axis) vs second-best prediction (y-axis) across 10,000 random cells for each predicted label from the scPRINT-2 corpus.}
    \label{fig-s3-heatmap-of-ethnicity-prediction-relationship-across-samples}
\end{figure}

\subsection{Heatmap of organism prediction relationship across samples}

\begin{figure}[H]
    \centering
    \includegraphics[width=6.09896in,height=6.09896in]{./supplementaries/scprint_2_media/image12.png}
    \caption[Heatmap of organism prediction relationship across samples.]{Generated using labels predicted as top-1 (x-axis) vs second-best prediction (y-axis) across 10,000 random cells for each predicted label from the scPRINT-2 corpus.}
    \label{fig-s4-heatmap-of-organism-prediction-relationship-across-samples}
\end{figure}

\subsection{Heatmap of organism prediction relationship using organism embedding similarity across samples}

\begin{figure}[H]
    \centering
    \includegraphics[width=6.5in,height=6.5in]{./supplementaries/scprint_2_media/image15.png}
    \caption[Heatmap of organism prediction relationship using organism embedding similarity across samples.]{Generated by averaging the embeddings for each predicted organism across 10,000 random cells for each predicted label in the scPRINT-2 corpus, and using the L2 distance.}
    \label{fig-s5-heatmap-of-organism-prediction-relationship-using-organism-embedding-similarity-across-samples}
\end{figure}

\subsection{Differential expression plots of the disagreeing cells between scPRINT-2 and ground truth}

\begin{figure}[H]
    \centering
    \includegraphics[width=6.25521in,height=2.63354in]{./supplementaries/scprint_2_media/image5.png}
    \caption[Differential expression plots of the disagreeing cells between scPRINT-2 and ground truth.]{Analysis on the cat/tiger cross-species dataset. ``pred failed'' is the macrophages labeled as type 2 pneumocytes by scPRINT-2.}
    \label{fig-s6-differential-expression-plots-of-the-disagreeing-cells-between-scprint-2-and-ground-truth}
\end{figure}

\subsection{Umap of the smart-seq dataset used in the varying context classification task}

\begin{figure}[H]
    \centering
    \includegraphics[width=6.5in,height=3.875in]{./supplementaries/scprint_2_media/image28.png}
    \caption[Umap of the smart-seq dataset used in the varying context classification task.]{Cortical areas smart-seq v4 dataset used in results section 2, showing Leiden clusters and ground truth cell types.}
    \label{fig-s7-umap-of-the-smart-seq-dataset-used-in-the-varying-context-classification-task}
\end{figure}

\subsection{Line plot of the classification across varying context length, using the most expressed genes}

\begin{figure}[H]
    \centering
    \includegraphics[width=5.93229in,height=3.64264in]{./supplementaries/scprint_2_media/image6.png}
    \caption[Line plot of the classification across varying context length, using the most expressed genes.]{Each new gene in context is 200 most expressed, then 500, 1000, etc. While cell types are often defined by their most expressed genes, and thus this doesn't change classification accuracy much, other, more complex labels continue increasing in accuracy as context length increases.}
    \label{fig-s8-line-plot-of-the-classification-across-varying-context-length-using-the-most-expressed-genes}
\end{figure}

\subsection{Illustration of the multiple perturbations applied to expression data in scPRINT-2}

\begin{figure}[H]
    \centering
    \includegraphics[width=5.10454in,height=3.15104in]{./supplementaries/scprint_2_media/image13.png}
    \caption[Illustration of the multiple perturbations applied to expression data in scPRINT-2.]{scPRINT can add noise and mask gene expression, modify the number of neighbors, and adjust context lengths.}
    \label{fig-s9-illustration-of-the-multiple-perturbations-applied-to-expression-data-in-scprint-2}
\end{figure}

\subsection{Distplot of the non-zero count distribution across cells from the three dataset qualities used}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image19.png}
    \caption[Distplot of the non-zero count distribution across cells from the three dataset qualities used.]{From left: good quality; center: excellent quality; right: poor quality datasets used in our denoising benchmark.}
    \label{fig-s10-distplot-of-the-non-zero-count-distribution-across-cells-from-the-three-dataset-qualities-used}
\end{figure}

\subsection{Umap over scPRINT-2 and PCA embeddings of the Xenium dataset}

\begin{figure}[H]
    \centering
    \includegraphics[width=6.42188in,height=3.98992in]{./supplementaries/scprint_2_media/image7.png}
    \caption[Umap over scPRINT-2 and PCA embeddings of the Xenium dataset.]{Left: raw PCA expression, right: scPRINT-2 embeddings with scPRINT-2 predicted cell types and diseases.}
    \label{fig-s11-umap-over-scprint-2-and-pca-embeddings-of-the-xenium-dataset}
\end{figure}

\subsection{Tangram mapping quality plots}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image32.png}
    \caption[Tangram mapping quality plots]{on the Xenium skin melanoma datasets and 10v3 skin melanoma datasets.}
    \label{fig-s12-tangram-mapping-quality-plots}
\end{figure}

\subsection{Illustration of scPRINT-2's generative imputation mechanism}

\begin{figure}[H]
    \centering
    \includegraphics[width=4.29788in,height=4.29788in]{./supplementaries/scprint_2_media/image31.png}
    \caption[Illustration of scPRINT-2's generative imputation mechanism.]{scPRINT encodes all 5000 measured genes into cell embeddings and decodes them on 5000 different unseen gene embeddings.}
    \label{fig-s13-illustration-of-scprint-2s-generative-imputation-mechanism}
\end{figure}

\subsection{Spatial plot of the Xenium melanoma dataset with scPRINT-2 predicted cell labels}

\begin{figure}[H]
    \centering
    \includegraphics[width=6.5in,height=5.61111in]{./supplementaries/scprint_2_media/image33.png}
    \caption[Spatial plot of the Xenium melanoma dataset with scPRINT-2 predicted cell labels]{for the disease, age, ethnicity, sex, and cell type labels on top of the selected Xenium skin melanoma patch.}
    \label{fig-s14-spatial-plot-of-the-xenium-melanoma-dataset-with-scprint-2-predicted-cell-labels}
\end{figure}

\subsection{Violin plot comparison of the gene's expression between predicted malignant vs the rest}

\begin{figure}[H]
    \centering
    \includegraphics[width=4.66146in,height=3.51402in]{./supplementaries/scprint_2_media/image26.png}
    \caption[Violin plot comparison of the gene's expression between predicted malignant vs the rest.]{BCL2, IGF1, EGFR, FGFR2, SOX10, key melanoma markers are highly expressed in the malignant cell type label group vs the rest, with a p-value of 10\textsuperscript{-234}}
    \label{fig-s15-violin-plot-comparison-of-the-genes-expression-between-predicted-malignant-vs-the-rest}
\end{figure}

\subsection{Differential expression plot of ``cancer'' disease labelled vs rest in the xenium dataset}

\begin{figure}[H]
    \centering
    \includegraphics[width=4.72396in,height=3.85546in]{./supplementaries/scprint_2_media/image25.png}
    \caption[Differential expression plot of ``cancer'' disease labelled vs rest in the xenium dataset.]{Cells whose disease label is ``cancer'' vs the rest in the Xenium skin melanoma dataset}
    \label{fig-s16-differential-expression-plot-of-cancer-disease-labelled-vs-rest-in-the-xenium-dataset}
\end{figure}

\subsection{Illustration of criss-cross attention}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image11.png}
    \caption[Illustration of criss-cross attention]{mechanism with sub-quadratic complexity}
    \label{fig-s17-illustration-of-criss-cross-attention}
\end{figure}

\subsection{Illustration of the similarity and dissimilarity-based contrastive losses used in scPRINT-2}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image21.png}
    \caption[Illustration of the similarity and dissimilarity-based contrastive losses used in scPRINT-2.]{The contrastive losses push embeddings from the same cell at different noise levels to be as similar as possible.}
    \label{fig-s18-illustration-of-the-similarity-and-dissimilarity-based-contrastive-losses-used-in-scprint-2}
\end{figure}

\subsection{Whisker plot of Open Problems' batch-integration with batch-correction-only scores}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image23.png}
    \caption[Whisker plot of Open Problems' batch-integration with batch-correction-only scores]{for scPRINT-1 and scPRINT-2 zero-shot, and finetuned, and all other models assessed in open problems.}
    \label{fig-s19-whisker-plot-of-open-problems-batch-integration-with-batch-correction-only-scores}
\end{figure}

\subsection{Whisker plot Open Problems' batch-integration with Bio-conservation-only scores}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image10.png}
    \caption[Whisker plot Open Problems' batch-integration with Bio-conservation-only scores]{for scPRINT-1 and scPRINT-2 zero-shot, and finetuned, and all other models assessed in open problems.}
    \label{fig-s20-whisker-plot-open-problems-batch-integration-with-bio-conservation-only-scores}
\end{figure}

\subsection{Umap of scPRINT-2's zero-shot multi-species expression embedding using the full cell-embedding}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image14.png}
    \caption[Umap of scPRINT-2's zero-shot multi-species expression embedding using the full cell-embedding]{from top to bottom, scPRINT-2 predicted cell type labels, ground truth cell type labels, and ground truth organism labels.}
    \label{fig-s21-umap-of-scprint-2s-zero-shot-multi-species-expression-embedding-using-the-full-cell-embedding}
\end{figure}

\subsection{Barplot of scIB score on scPRINT-2's multi-species integration}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image29.png}
    \caption[Barplot of scIB score on scPRINT-2's multi-species integration]{showing total, bio conservation, and batch integration across scPRINT-2 zero-shot, and fine-tuned version using both the full cell-embedding and cell-type-only cell-embedding}
    \label{fig-s22-barplot-of-scib-score-on-scprint-2s-multi-species-integration}
\end{figure}

\subsection{Umap of scPRINT-2's zero-shot multi-species expression embedding using the cell-type cell-embedding}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image8.png}
    \caption[Umap of scPRINT-2's zero-shot multi-species expression embedding using the cell-type cell-embedding]{from top to bottom, scPRINT-2 predicted cell type labels, ground truth cell type labels, and ground truth organism labels.}
    \label{fig-s23-umap-of-scprint-2s-zero-shot-multi-species-expression-embedding-using-the-cell-type-cell-embedding}
\end{figure}

\subsection{Umap of scPRINT-2's multi-species expression embedding post-finetuning using the full cell-embedding}

\begin{figure}[H]
    \centering
    \includegraphics[width=6.09198in,height=3.33955in]{./supplementaries/scprint_2_media/image34.png}
    \caption[Umap of scPRINT-2's multi-species expression embedding post-finetuning using the full cell-embedding]{from left to right and top to bottom, ground truth cell type, scPRINT-2 predicted cell type labels, ground truth organism labels, and Leiden clusters.}
    \label{fig-s24-umap-of-scprint-2s-multi-species-expression-embedding-post-finetuning-using-the-full-cell-embedding}
\end{figure}

\subsection{Differential expression plot of the human vs mouse dataset from section 4}

\begin{figure}[H]
    \centering
    \includegraphics[width=6.12385in,height=4.92188in]{./supplementaries/scprint_2_media/image9.png}
    \caption[Differential expression plot of the human vs mouse dataset from section 4.]{Rest is mouse here.}
    \label{fig-s25-differential-expression-plot-of-the-human-vs-mouse-dataset-from-section-4}
\end{figure}

\subsection{Over-representation plot of humanized mouse data vs real mouse data compared to human}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image3.png}
    \caption[Over-representation plot of humanized mouse data vs real mouse data compared to human.]{Differentially expressed genes in scPRINT-2's humanized mouse data vs real mouse data compared to human.}
    \label{fig-s26-over-representation-plot-of-humanized-mouse-data-vs-real-mouse-data-compared-to-human}
\end{figure}

\subsection{Over-representation plot of female-like male data vs real female data compared to male}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image2.png}
    \caption[Over-representation plot of female-like male data vs real female data compared to male.]{Top differentially expressed genes in scPRINT-2's female-like male data vs real female data compared to male.}
    \label{fig-s27-over-representation-plot-of-female-like-male-data-vs-real-female-data-compared-to-male}
\end{figure}

\subsection{Dot Plot of Gene-set enrichment analysis over the differential expression analysis of section 4}

\begin{figure}[H]
    \centering
    \includegraphics[width=6.11529in,height=2.41518in]{./supplementaries/scprint_2_media/image35.png}
    \caption[Dot Plot of Gene-set enrichment analysis over the differential expression analysis of section 4.]{Showing the top 10 most enriched gene sets from the GO molecular function 2023 database.}
    \label{fig-s28-dot-plot-of-gene-set-enrichment-analysis-over-the-differential-expression-analysis-of-section-4}
\end{figure}

\subsection{Output gene embedding for a non-fully trained model without XPressor architecture}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image18.png}
    \caption[Output gene embedding for a non-fully trained model without XPressor architecture.]{Overlaying in color, from left to right, the Leiden clusters, the expression values, and the zero vs non-zero expression. Despite displaying multiple clusters, the number of enriched pathways in each is still smaller than for a model using XPressor. (see Figure 5)}
    \label{fig-s29-output-gene-embedding-for-a-non-fully-trained-model-without-xpressor-architecture}
\end{figure}

\subsection{Venn diagram of the different ground truth gene networks}

\begin{figure}[H]
    \centering
    \includegraphics[width=4.82813in,height=4.82813in]{./supplementaries/scprint_2_media/image22.png}
    \caption[Venn diagram of the different ground truth gene networks]{showing overlap in the edges using gene symbols over the five ground truths used in our benchmark}
    \label{fig-s30-venn-diagram-of-the-different-ground-truth-gene-networks}
\end{figure}

\subsection{Whisker plot of AUPRC-ratio scores for scPRINT-1 and scPRINT-2}

\begin{figure}[H]
    \centering
    \includegraphics[width=6.38646in,height=3.89163in]{./supplementaries/scprint_2_media/image27.png}
    \caption[Whisker plot of AUPRC-ratio scores for the benchmark of scPRINT-1 vs scPRINT-2]{using their respective GRN-extraction methods, showing that the scPRINT-2 extraction, while highlighting more relevant top connections, remains relatively similar to the scPRINT-1 version on the AUPRC-ratio scores on each of the six ground truth networks.}
    \label{fig-s31-whisker-plot-of-auprc-ratio-scores-for-scprint-1-and-scprint-2}
\end{figure}

\subsection{Additional scPRINT-2 generated gene network computed from CDC45}

\begin{figure}[H]
    \centering
    \includegraphics[width=5.91146in,height=5.91146in]{./supplementaries/scprint_2_media/image16.png}
    \caption[Additional scPRINT-2 generated gene network computed from CDC45.]{Subpart of the gene network using CDC45 as a seed gene and computed on 1024 mouse macrophages, showing how these networks can exhibit complex structures.}
    \label{fig-s32-additional-scprint-2-generated-gene-network-computed-from-cdc45}
\end{figure}
