\subsection{Detailed version of the additive
benchmark}\label{table-s1-detailed-version-of-the-additive-benchmark}

\includegraphics[width=\textwidth]{./supplementaries/scprint_2_media/image4.png}

Detailed version of the additive benchmark, listing every value.

\subsection{Detailed scIB biological conservation scores on
the xenium
dataset}\label{table-s3-detailed-scib-biological-conservation-scores-on-the-xenium-dataset}

\includegraphics[width=\textwidth]{./supplementaries/scprint_2_media/image30.png}

scPRINT vs PCA on expression. ScPRINT performs better, likely by
denoising the expression.

\subsection{Detailed scIB scores on the unseen species
integration
task}\label{table-s4-detailed-scib-scores-on-the-unseen-species-integration-task}

\includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image20.png}

Details of the full scIB results comparing no integration, random
embeddings sampled from the multivariate Gaussian, and different
versions of scPRINT zero-shot or fine-tuned, using the merged embeddings
or the cell type ones

\section{Supplementary figures for scPRINT-2}
\labels{supplementary-figures-2}

\subsection{Illustration of the full scPRINT-2's architecture,
input, and
output}\label{fig-s1-illustration-of-the-full-scprint-2s-architecture-input-and-output}

\includegraphics[width=5.92104in,height=6.85698in]{./supplementaries/scprint_2_media/image24.png}

In-depth illustration of the full scPRINT-2's architecture, input, and
output with all its main different components and the data flow.

\subsection{Barplot of the F1-macro scores on the
label-projection task of the Open Problem
benchmark}\label{fig-s2-barplot-of-the-f1-macro-scores-on-the-label-projection-task-of-the-open-problem-benchmark}

\includegraphics[width=6.22396in,height=4.54658in]{./supplementaries/scprint_2_media/image17.png}

Comparison of scPRINT-1 and scPRINT-2, zero-shot and finetuned, with all
other tested methods in Open Problems.

\subsection{Heatmap of ethnicity prediction relationship across
samples}\label{fig-s3-heatmap-of-ethnicity-prediction-relationship-across-samples}

\includegraphics[width=6.39163in,height=6.39163in]{./supplementaries/scprint_2_media/image1.png}

It is generated using labels predicted as top-1 (x-axis) vs second-best
prediction (y-axis) across 10,000 random cells for each predicted label
from the scPRINT-2 corpus.

\subsection{Heatmap of organism prediction relationship across
samples}\label{fig-s4-heatmap-of-organism-prediction-relationship-across-samples}

\includegraphics[width=6.09896in,height=6.09896in]{./supplementaries/scprint_2_media/image12.png}

It is generated using labels predicted as top-1 (x-axis) vs second-best
prediction (y-axis) across 10,000 random cells for each predicted label
from the scPRINT-2 corpus.

\subsection{Heatmap of organism prediction relationship using
organism embedding similarity across
samples}\label{fig-s5-heatmap-of-organism-prediction-relationship-using-organism-embedding-similarity-across-samples}

\includegraphics[width=6.5in,height=6.5in]{./supplementaries/scprint_2_media/image15.png}

It is generated by averaging the embeddings for each predicted organism
across 10,000 random cells for each predicted label in the scPRINT-2
corpus, and using the L2 distance.

\subsection{Differential expression plots of the disagreeing
cells between scPRINT-2 and ground
truth}\label{fig-s6-differential-expression-plots-of-the-disagreeing-cells-between-scprint-2-and-ground-truth}

\includegraphics[width=6.25521in,height=2.63354in]{./supplementaries/scprint_2_media/image5.png}

The differential expression is made on the cat/tiger cross-species
dataset. ``pred failed'' is the macrophages labeled as type 2
pneumocytes by scPRINT-2.

\subsection{Umap of the smart-seq dataset used in the varying
context classification
task}\label{fig-s7-umap-of-the-smart-seq-dataset-used-in-the-varying-context-classification-task}

\includegraphics[width=6.5in,height=3.875in]{./supplementaries/scprint_2_media/image28.png}

Umap of the cortical areas smart-seq v4 dataset used in the varying
context classification task in results section 2, showing Leiden
clusters and ground truth cell types.

\subsection{Line plot of the classification across varying
context length, using the most expressed
genes}\label{fig-s8-line-plot-of-the-classification-across-varying-context-length-using-the-most-expressed-genes}

\includegraphics[width=5.93229in,height=3.64264in]{./supplementaries/scprint_2_media/image6.png}

On the same dataset, but this time using the most expressed genes.
Meaning each new gene in context is 200 most expressed, then 500, 1000,
etc. We can see that while cell types are often defined by their most
expressed genes, and thus this doesn't change classification accuracy
much, other, more complex labels continue increasing in accuracy as
context length increases.

\subsection{Illustration of the multiple
perturbations applied to expression data in scPRINT-2
}\label{fig-s9-illustration-of-the-multiple-perturbations-applied-to-expression-data-in-scprint-2}

\includegraphics[width=5.10454in,height=3.15104in]{./supplementaries/scprint_2_media/image13.png}

scPRINT can add noise and mask gene expression, modify the
number of neighbors, and adjust context
lengths.

\subsection{Distplot of the non-zero count distribution across
cells from the three dataset qualities
used}\label{fig-s10-distplot-of-the-non-zero-count-distribution-across-cells-from-the-three-dataset-qualities-used}

\includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image19.png}

Non-zero count distributions across cells from left: good quality;
center: excellent quality; right: poor quality datasets used in our
denoising benchmark.

\subsection{Umap over scPRINT-2 and PCA
embeddings of the Xenium dataset
}\label{fig-s11-umap-over-scprint-2-and-pca-embeddings-of-the-xenium-dataset}

\includegraphics[width=6.42188in,height=3.98992in]{./supplementaries/scprint_2_media/image7.png}

Umap of left: raw PCA expression, right: scPRINT-2 embeddings with
scPRINT-2 predicted cell types and diseases.

\subsection{Tangram mapping quality
plots}\label{fig-s12-tangram-mapping-quality-plots}

\includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image32.png}

Tangram mapping quality plots on the Xenium skin melanoma datasets and
10v3 skin melanoma datasets.

\subsection{Illustration of scPRINT-2's generative imputation
mechanism}\label{fig-s13-illustration-of-scprint-2s-generative-imputation-mechanism}

\being{center}
\includegraphics[width=4.29788in,height=4.29788in]{./supplementaries/scprint_2_media/image31.png}
\end{center}

scPRINT encodes all 5000 measured genes into cell embeddings and decodes
them on 5000 different unseen gene embeddings.

\subsection{Spatial plot of the Xenium melanoma dataset with
scPRINT-2 predicted cell
labels}\label{fig-s14-spatial-plot-of-the-xenium-melanoma-dataset-with-scprint-2-predicted-cell-labels}

\includegraphics[width=6.5in,height=5.61111in]{./supplementaries/scprint_2_media/image33.png}

scPRINT-2 predicted cell labels for the disease, age, ethnicity, sex,
and cell type labels on top of the selected Xenium skin melanoma patch.

\subsection{Violin plot comparison of the gene's expression
between predicted malignant vs the
rest}\label{fig-s15-violin-plot-comparison-of-the-genes-expression-between-predicted-malignant-vs-the-rest}

\begin{center}
\includegraphics[width=4.66146in,height=3.51402in]{./supplementaries/scprint_2_media/image26.png}
\end{center}

Violin plot showing that BCL2, IGF1, EGFR, FGFR2, SOX10, key melanoma
markers are highly expressed in the malignant cell type label group vs
the rest, with a p-value of 10\textsuperscript{-234}

\subsection{Differential expression plot of ``cancer'' disease
labelled vs rest in the xenium
dataset}\label{fig-s16-differential-expression-plot-of-cancer-disease-labelled-vs-rest-in-the-xenium-dataset}


\begin{center}
\includegraphics[width=4.72396in,height=3.85546in]{./supplementaries/scprint_2_media/image25.png}
\end{center}

Differential expression plot of cells whose disease label is ``cancer''
vs the rest in the Xenium skin melanoma dataset

\subsection{Illustration of criss-cross
attention}\label{fig-s17-illustration-of-criss-cross-attention}

\includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image11.png}

Illustration of our sub-quadratic complexity criss-cross attention
mechanism

\subsection{Illustration of the similarity and
dissimilarity-based contrastive losses used in
scPRINT-2}\label{fig-s18-illustration-of-the-similarity-and-dissimilarity-based-contrastive-losses-used-in-scprint-2}

\includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image21.png}

The contrastive losses push embeddings from the same cell at different
noise levels to be as similar as possible.

\subsection{Whisker plot of Open Problems' batch-integration
with batch-correction-only
scores}\label{fig-s19-whisker-plot-of-open-problems-batch-integration-with-batch-correction-only-scores}

\includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image23.png}

Open Problems' batch-integration with batch-correction-only scores for
scPRINT-1 and scPRINT-2 zero-shot, and finetuned, and all other models
assessed in open problems.

\subsection{Whisker plot Open Problems' batch-integration with
Bio-conservation-only
scores}\label{fig-s20-whisker-plot-open-problems-batch-integration-with-bio-conservation-only-scores}

\includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image10.png}

Open Problems' batch-integration with Bio-conservation-only scores for
scPRINT-1 and scPRINT-2 zero-shot, and finetuned, and all other models
assessed in open problems.

\subsection{Umap of scPRINT-2's zero-shot
multi-species expression embedding using the full cell-embedding
}\label{fig-s21-umap-of-scprint-2s-zero-shot-multi-species-expression-embedding-using-the-full-cell-embedding}

\includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image14.png}

scPRINT-2's zero-shot multi-species expression embedding using the full
cell-embedding from top to bottom, scPRINT-2 predicted cell type labels,
ground truth cell type labels, and ground truth organism labels.

\subsection{Barplot of scIB score on scPRINT-2's multi-species
integration}\label{fig-s22-barplot-of-scib-score-on-scprint-2s-multi-species-integration}

\includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image29.png}

showing total, bio conservation, and batch integration across scPRINT-2
zero-shot, and fine-tuned version using both the full cell-embedding and
cell-type-only cell-embedding

\subsection{Umap of scPRINT-2's zero-shot
multi-species expression embedding using the cell-type cell-embedding
}\label{fig-s23-umap-of-scprint-2s-zero-shot-multi-species-expression-embedding-using-the-cell-type-cell-embedding}

\includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image8.png}

scPRINT-2's zero-shot multi-species expression embedding using the
cell-type cell-embedding from top to bottom, scPRINT-2 predicted cell
type labels, ground truth cell type labels, and ground truth organism
labels.

\subsection{Umap of scPRINT-2's multi-species expression
embedding post-finetuning using the full
cell-embedding}\label{fig-s24-umap-of-scprint-2s-multi-species-expression-embedding-post-finetuning-using-the-full-cell-embedding}

\includegraphics[width=6.09198in,height=3.33955in]{./supplementaries/scprint_2_media/image34.png}

scPRINT-2's multi-species expression embedding post-finetuning using the
full cell-embedding from left to right and top to bottom, ground truth
cell type, scPRINT-2 predicted cell type labels, ground truth organism
labels, and Leiden clusters.

\subsection{Differential expression plot of the human vs mouse
dataset from section
4}\label{fig-s25-differential-expression-plot-of-the-human-vs-mouse-dataset-from-section-4}

\includegraphics[width=6.12385in,height=4.92188in]{./supplementaries/scprint_2_media/image9.png}

Differential expression plot of the human vs mouse dataset from section
4. Rest is mouse here.

\subsection{Over-representation plot of
humanized mouse data vs real mouse data compared to human
}\label{fig-s26-over-representation-plot-of-humanized-mouse-data-vs-real-mouse-data-compared-to-human}

\includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image3.png}

Over-representation plot of differentially expressed genes in
scPRINT-2's humanized mouse data vs real mouse data compared to human.

\subsection{Over-representation plot of female-like male data
vs real female data compared to
male}\label{fig-s27-over-representation-plot-of-female-like-male-data-vs-real-female-data-compared-to-male}

\includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image2.png}

Over-representation plot of top differentially expressed genes in
scPRINT-2's female-like male data vs real female data compared to male.

\subsection{Dot Plot of Gene-set enrichment analysis over the
differential expression analysis of section
4}\label{fig-s28-dot-plot-of-gene-set-enrichment-analysis-over-the-differential-expression-analysis-of-section-4}

\includegraphics[width=6.11529in,height=2.41518in]{./supplementaries/scprint_2_media/image35.png}

Showing the top 10 most enriched gene sets from the GO molecular
function 2023 database.

\subsection{Output gene embedding for a non-fully trained model
without XPressor
architecture}\label{fig-s29-output-gene-embedding-for-a-non-fully-trained-model-without-xpressor-architecture}

\includegraphics[width=\linewidth]{./supplementaries/scprint_2_media/image18.png}

Overlaying in color, from left to right, the Leiden clusters, the
expression values, and the zero vs non-zero expression. Despite
displaying multiple clusters, the number of enriched pathways in each is
still smaller than for a model using XPressor. (see Figure 5)

\subsection{Venn diagram of the different ground truth gene
networks}\label{fig-s30-venn-diagram-of-the-different-ground-truth-gene-networks}

\includegraphics[width=4.82813in,height=4.82813in]{./supplementaries/scprint_2_media/image22.png}

Venn diagram of the different ground truth gene networks showing overlap
in the edges using gene symbols over the five ground truths used in our
benchmark

\subsection{Whisker plot of AUPRC-ratio scores for scPRINT-1
and
scPRINT-2}\label{fig-s31-whisker-plot-of-auprc-ratio-scores-for-scprint-1-and-scprint-2}

\includegraphics[width=6.38646in,height=3.89163in]{./supplementaries/scprint_2_media/image27.png}

Whisker plot of AUPRC-ratio scores for the benchmark of scPRINT-1 vs
scPRINT-2 using their respective GRN-extraction methods, showing that
the scPRINT-2 extraction, while highlighting more relevant top
connections, remains relatively similar to the scPRINT-1 version on the
AUPRC-ratio scores on each of the six ground truth networks.

\subsection{Additional scPRINT-2 generated gene
network computed from CDC45
}\label{fig-s32-additional-scprint-2-generated-gene-network-computed-from-cdc45}

\includegraphics[width=5.91146in,height=5.91146in]{./supplementaries/scprint_2_media/image16.png}

Subpart of the scPRINT-2 generated gene network using CDC45 as a seed
gene and computed on 1024 mouse macrophages, showing how these networks
can exhibit complex structures.
